src/save_options/registration.py

```diff
@@ -0,0 +1,111 @@
+import contextlib
+import inspect
+from collections.abc import Callable, Sequence
+from typing import Any
+
+from unrealsdk import logging
+
+from mods_base import BaseOption, GroupedOption, Mod, NestedOption
+from save_options.options import SaveOption
+
+# Allowing GroupedOption and NestedOption if all their children are SaveOption
+ModSaveOptions = dict[str, BaseOption]
+registered_save_options: dict[str, ModSaveOptions] = {}
+
+SaveCallback = Callable[[], None]
+save_callbacks: dict[str, SaveCallback] = {}
+
+LoadCallback = Callable[[], None]
+load_callbacks: dict[str, LoadCallback] = {}
+
+registered_mods: dict[str, Mod] = {}
+
+
+def _treat_as_save_option(obj: Any) -> bool:
+    if isinstance(obj, SaveOption):
+        return True
+    if isinstance(obj, GroupedOption | NestedOption):
+        if all(_treat_as_save_option(child) for child in obj.children):
+            return True
+        logging.error(
+            f"Option {obj.identifier} has both regular BaseOption and SaveOption"
+            f"defined as children. SaveOption instances will be ignored"  # noqa: COM812
+        )
```

Comment:
This should be dev warning, not error. A user who sees this is going to have no idea what the problem is or what they should be doing about it.

Second line is missing a leading space and trailing full stop (and COM812).